File: staircase.py
#!/usr/bin/python3


def staircase(n):
    level_string = ""
    for i in range(n):
        level_string += " " * (n - 1 - i)
        level_string += "#" * (i + 1)
        level_string += "\n"
    print(level_string)
    return level_string


staircase(6)

File: staircase.go
package main

import "fmt"

// O(2n^2)
func staircase(n int) (levelString string) {
	for level := 0; level < n; level++ {
		// can also use strings.Repeat() for spaces and hashes
		numberOfSpaces := n - 1 - level
		for space := 0; space < numberOfSpaces; space++ {
			levelString += " "
		}
		numberOfHashes := level + 1
		for hash := 0; hash < numberOfHashes; hash++ {
			levelString += "#"
		}
		levelString += "\n"
	}
	fmt.Println(levelString)
	return
}

func main() {
	staircase(6)
}
